pkg/templates/readsecret/template.go

```diff
@@ -0,0 +1,40 @@
+package readsecret
+
+import (
+	"fmt"
+
+	"golang.stackrox.io/kube-linter/pkg/check"
+	"golang.stackrox.io/kube-linter/pkg/config"
+	"golang.stackrox.io/kube-linter/pkg/diagnostic"
+	"golang.stackrox.io/kube-linter/pkg/objectkinds"
+	"golang.stackrox.io/kube-linter/pkg/templates"
+	"golang.stackrox.io/kube-linter/pkg/templates/readsecret/internal/params"
+	"golang.stackrox.io/kube-linter/pkg/templates/util"
+	v1 "k8s.io/api/core/v1"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Read Secret From Environment Variables",
```

Comment:
Hmm, not sure I understand. Using `SecretKeyRef` is the recommended practice. Why do we want to forbid it?

Reply:
This template implements the following CIS benchmark. Using secrets as environment variables is not recommended as the environment variables are likely be logged and therefore expose secrets to log viewers.

**5.4.1 Prefer using secrets as files over secrets as environment variables (Manual)**

Profile Applicability:
• Level 2 - Master Node Description:

> Kubernetes supports mounting secrets as data volumes or as environment variables. Minimize the use of environment variable secrets.

Rationale:

> It is reasonably common for application code to log out its environment (particularly in the event of an error). This will include any secret values passed in as environment variables, so secrets can easily be exposed to any user or entity who has access to the logs.

Audit:

> Run the following command to find references to objects which use environment variables defined from secrets.
` kubectl get all -o jsonpath='{range .items[?(@..secretKeyRef)]} {.kind} {.metadata.name} {"\n"}{end}' -A`

Remediation:

> If possible, rewrite application code to read secrets from mounted secret files, rather than from environment variables.

Reply:
Okay, got it. Fair enough.